### ISSUE TITLE
Global Search by Service Name in Web UI

### ISSUE THREAD

**Issue by mahomedh** [netbox-community/netbox#4643]
### Environment
* Python version:  3.6.9
* NetBox version:  2.8.4

### Proposed Functionality
Globally search for all Devices and VMs that have a service name that matches search parameter. For example if you search for `SSH` it lists all devices with a matching service name.

### Use Case
I'll personally be using the services to record all A Records that are associated with the particular Web Server device/VM. But this is not very much use if you can't search for devices with the assigned address.

Another use case would be if you want to search for all servers that have a particular service assigned (like postgresql for example).

### Database Changes
None (afaik)

### External Dependencies
None (afaik)

**Comment by jeremystretch** [comment]
> I'll personally be using the services to record all A Records that are associated with the particular Web Server device/VM.

This is not a valid use of the service model. A service is a running process on a device that is associated with a specific protocol (TCP/UDP) and port number (and optionally one or more IP addresses assigned to the device). It is intended to model available services and is not suitable for tracking DNS records.

**Comment by mahomedh** [comment]
I agree it's not the intended design, but it works for our use case as we have upto 90 records on some servers. The DNS and custom text fields cannot accomodate this many records. Currently we keep this in a text hosts file. The idea is to be able to store it in Netbox and build the hosts file from data stored in Netbox.

Regardless though, would you not agree it would still be useful to search by service name?

**Comment by jeremystretch** [comment]
I don't think it makes sense to search for devices by the name of an assigned service. That seems prone to returning too many irrelevant results. For example, you might be looking for devices with "dns" in their name but end up with all devices owning a "DNS" service.

**Comment by mahomedh** [comment]
I see your concern. It doesn't matter much now because I found that the "Name" field is too short for some our aliases. Instead, I now create the following:

```
Name          Protocol    Port      IP                   Description
DNS Alias     TCP         53        192.168.0.10/24      www.mydomain.com
```

 I have also just discovered that I can go to IPAM > Services and I can search there. So although it's not as convenient, it makes it usable for our use case.

So happy to close this. Thanks.
